### PR TITLE
chore: Ability to run in local cluster without pushing images

### PR DESCRIPTION
**Pull request by whynowy**
`make deploy` to run the controller in a local cluster without pushing images.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,13 +1,16 @@
+# Setting SHELL to bash allows bash commands to be executed by recipes.
+# Options are set to exit when a recipe line exits non-zero or a piped command fails.
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
 
 # Image URL to use all building/pushing image targets
 IMG ?= numaplane-controller
 VERSION ?= latest
+BASE_VERSION := latest
```
Comment (whynowy) [inline Makefile:9]: Will be useful for future release scripts.

File: config/manager/config.yaml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: numaplane-controller-config
+  name: controller-config
   namespace: numaplane-system
 data:
   config.yaml: |
-    clusterName: "CLUSTER_NAME_VALUE" # Replace with the cluster name where numaplane will get deployed.
+    clusterName: CLUSTER_NAME_VALUE # Replace with the cluster name where numaplane will get deployed.
```
Comment (whynowy) [inline config/manager/config.yaml:8]: The extra `"` was making trouble.

File: config/manager/manager.yaml [hunk 3]
```diff
@@ -103,4 +103,4 @@ spec:
       volumes:
         - name: config-volume
           configMap:
-            name: numaplane-controller-config
+            name: controller-config
```
Comment (whynowy) [inline config/manager/manager.yaml:106]: Having the prefix made the configmap name `numaplane-numaplane-controller-config`.

File: Makefile [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
 # Default cluster name where numaplane get deployed, update it as needed.
-CLUSTER_NAME ?= "staging-usw2-k8s"
-#TODO: add back once ready to push to quay.io
-#IMAGE_NAMESPACE ?= quay.io/numaproj
-#IMAGE_FULL_PATH ?= $(IMAGE_NAMESPACE)/$(IMG):$(VERSION)
-IMAGE_FULL_PATH ?= $(IMG):$(VERSION)
+CLUSTER_NAME ?= staging-usw2-k8s
```
Comment (whynowy) [inline Makefile:11]: The `"` was making trouble.

File: Makefile [hunk 5]
```diff
@@ -146,9 +160,8 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified
 	$(KUSTOMIZE) build config/crd | $(KUBECTL) delete --ignore-not-found=$(ignore-not-found) -f -
 
 .PHONY: deploy
-deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
-	cd config/manager && $(KUSTOMIZE) edit set image numaplane-controller=${IMAGE_FULL_PATH}
```
Comment (whynowy) [inline Makefile:150]: Should not use `kustomize edit` here, it updates `kustomization.yaml`.

File: Makefile [hunk 6]
```diff
@@ -1,13 +1,16 @@
+# Setting SHELL to bash allows bash commands to be executed by recipes.
+# Options are set to exit when a recipe line exits non-zero or a piped command fails.
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
 
 # Image URL to use all building/pushing image targets
 IMG ?= numaplane-controller
 VERSION ?= latest
```
Comment (xdevxy) [inline Makefile:8]: What is the difference between `VERSION` and `BASE_VERSION`?

**Review by xdevxy — COMMENTED** [review]
Did you also try `make deploy` with image on remote registry is still working?

**Comment by whynowy** [comment]
> Did you also try `make deploy` with image on remote registry is still working?

This command is supposed to be used for development in local cluster only, in that case, we should not ask the developers to use a remote registry. If we want to deploy a remote image to a cluster (no matter if it's a local cluster or a remote one), for example, assume we have an official released version, and we want to verify if that release is good, then we should follow the released instruction to install (e.g. kubectl apply -f <released-manifests-url>).

File: Makefile (hunk 6 above)
Comment (juliev0) [reply Makefile:8]: would it make sense to add a comment for it then?

Comment (whynowy) [reply Makefile:8]: Added a short comment for that.

**Comment by xdevxy** [comment]
> > Did you also try `make deploy` with image on remote registry is still working?
> 
> This command is supposed to be used for development in local cluster only, in that case, we should not ask the developers to use a remote registry. If we want to deploy a remote image to a cluster (no matter if it's a local cluster or a remote one), for example, assume we have an official released version, and we want to verify if that release is good, then we should follow the released instruction to install (e.g. kubectl apply -f ).

Got it, makes sense, thanks!

**Review by xdevxy — APPROVED** [review]

**Pull request merged**